### PR TITLE
refactor: make zipWithIndex and enumerator consistent with mapWithIndex (issue #5080)

### PR DESCRIPTION
**Pull request by stephentetley**
This PR changes the argument orders of `zipWithIndex` and `enumerator` so they match `mapWithIndex` - see issue #5080

I think `zipWithIndex` with index first is a good improvement over the status quo - a list of (index, value) pairs seems consistent with how we represent other data e.g. Map (key, value) pairs rather than (value, index) pairs.

Similarly a `foreach` comprehension with an `enumerator` seems more consistent with a traditional for-loop if the loop variables are in (index, value) order.

### CODE REVIEW

File: main/src/library/Chain.flix [hunk 1]
```diff
@@ -911,10 +911,10 @@ namespace Chain {
     /// Returns a chain where each element `e` is mapped to `(e, i)` where `i`
```
Comment (JonathanStarup) [inline main/src/library/Chain.flix:911]: The doc comment is outdated here

File: main/src/library/DelayList.flix [hunk 2]
```diff
@@ -1162,11 +1162,11 @@ namespace DelayList {
     ///
```
Comment (JonathanStarup) [inline main/src/library/DelayList.flix:1162]: Heres also a comment

File: main/src/library/List.flix [hunk 3]
```diff
@@ -930,10 +930,10 @@ namespace List {
     /// Returns a list where each element `e` is mapped to `(e, i)` where `i`
```
Comment (JonathanStarup) [inline main/src/library/List.flix:930]: Here is also a comment

File: main/src/library/Nec.flix [hunk 4]
```diff
@@ -674,10 +674,10 @@ namespace Nec {
     /// Returns a Nec where each element `e` is mapped to `(e, i)` where `i`
```
Comment (JonathanStarup) [inline main/src/library/Nec.flix:674]: ^

File: main/src/library/Nel.flix [hunk 5]
```diff
@@ -526,13 +526,13 @@ namespace Nel {
     /// Returns a new non-empty list where each element `e` is mapped to `(e, i)`
```
Comment (JonathanStarup) [inline main/src/library/Nel.flix:526]: ^

**Review by JonathanStarup — CHANGES_REQUESTED** [review]
I found some comments that reference the pair order

**Comment by stephentetley** [comment]
Thanks for the review Jonathan - I'll get those comments sorted today.

File: main/src/library/DelayList.flix (hunk 2 above)
Comment (stephentetley) [reply main/src/library/DelayList.flix:1162]: fixed

File: main/src/library/List.flix (hunk 3 above)
Comment (stephentetley) [reply main/src/library/List.flix]: fixed

File: main/src/library/Nec.flix (hunk 4 above)
Comment (stephentetley) [reply main/src/library/Nec.flix]: fixed

File: main/src/library/Nel.flix (hunk 5 above)
Comment (stephentetley) [reply main/src/library/Nel.flix]: fixed

**Review by JonathanStarup — APPROVED** [review]

**Comment by magnus-madsen** [comment]
Thanks guys!

**Pull request merged**